### ISSUE TITLE
Make Java object instance tied to a V8Object.

### ISSUE THREAD

**Issue by skmll** [eclipsesource/J2V8#137]
The title probably can not express the idea i want to, but imagine the following situation:

I have a v8Object for each instance of a class. In that v8Object i register some functions of said Java instance, for example:

`public class Vehicle {`
	
	private V8Object v8Object;

	public Vehicle(V8Object v8Object) {
		this.v8Object = v8Object;
        this.v8Object.registerJavaMethod(this, "setAttributes", "setAttributes", new Class<?>[]{V8Object.class});
	}

	public void setAttributes(V8Object object) {
		....
	}
}

Then I am returning that v8Object to the runtime, something like:

public class VehicleMaker {

	private V8 runtime;

	public VehicleMaker(V8 runtime) {
		this.runtime = runtime;
        V8Object v8VMaker = new V8Object(runtime);
        runtime.add("VehicleMaker", v8VMaker);

        v8VMaker.registerJavaMethod(this, "makeVehicle", "makeVehicle", null);
	}

	public V8Object makeVehicle() {
		V8Object v8Vehicle = new V8Object(this.runtime);
		Vehicle v = new Vehicle(v8Vehicle);

		return v8Vehicle;
	}
}

So in javascript, we can do something like:

	var newVehicle = VehicleMaker.makeVehicle();
	newVehicle.setAttributes({...});

But somewhere in the future, I want to be able to do: (also in javascript)

	me.purchaseVehicle(newVehicle); // -> the vehicle created above

The problem is: assuming that "purchaseVehicle" is also a java registed method,
there is no way (or I havent found it) to tie the V8Object (that I get as 
argument to the "purchaseVehicle" mehtod) to the initial Vehicle Java instance that 
I created and registed the "setAttributes" method.

Actually, let me rephrase it: there is no good way. I have been able to register
an integer constant in the v8Object, and put the Vehicle instance in a Map where 
the key is that integer. That way when I get the V8Object in the "purchaseVehicle"
method I can map it to the Java Vehicle instance.

So, I was wondering if there is a cleaner way to do this, since this way I get 
a giant Map with instances that may be preventing the GC from cleaning something.
I followed the path to registing a method reflectively, and I see the object
where the method is implemented is stored somewhere in a methodDescriptor, but it
seems like I cant access that either.

Anyway, thanks a lot, and sorry if this isn't the right place for this question.

PS: Haven´t tested the code or anything, its just an example of what I want to do.

**Comment by Shujito** [comment]
@skmll did you find a solution?
